### ISSUE TITLE
Unsupported warning for browser not in browserslist query

### ISSUE THREAD

**Issue by einsiol** [amilajack/eslint-plugin-compat#252]
After updating from `v2.7.0` to `v3.3.0`, I'm getting error messages about a feature not supported in browsers that I don't intend to support.

The errors are these:
```
/scripts/component/create-component.js
  12:9  error  Promise.all() is not supported in Opera Mini all, IE Mobile 11, Blackberry Browser 10  compat/compat

/scripts/svg/index.js
  38:9  error  Promise.all() is not supported in Opera Mini all, IE Mobile 11, Blackberry Browser 10  compat/compat

/src/index.js
  14:1  error  Promise.all() is not supported in Opera Mini all, IE Mobile 11, Blackberry Browser 10  compat/compat

✖ 5 problems (5 errors, 0 warnings)
```

My setup for **browserslist** in the `package.json` is this:
```javascript
"browserslist": [
    "last 2 versions",
    "not android < 100",
    "not and_qq < 100",
    "not and_uc < 100",
    "not baidu < 100",
    "not bb < 100",
    "not ie <= 11",
    "not ie_mob <= 11",
    "not opera < 1000",
    "not op_mini all",
    "not op_mob < 100",
    "not samsung < 100"
  ]
```

And the results from `npx browserslist` is:
```
and_chr 71
and_ff 64
chrome 72
chrome 71
edge 18
edge 17
firefox 65
firefox 64
ios_saf 12.0-12.1
ios_saf 11.3-11.4
safari 12
safari 11.1
```

So according to the list above from `npx browserslist`, unless I'm missing something, I'm sure I should not be getting these linting errors from **eslint-plugin-compat**.

**Comment by maoberlehner** [comment]
Same here:

```
error  Array.from() is not supported in IE 10  compat/compat
```

```
npx browserlist
and_chr 80
and_uc 12.12
chrome 80
chrome 79
edge 18
firefox 73
firefox 72
ie 11
ios_saf 13.3
ios_saf 12.2-12.4
safari 13
samsung 10.1
```

**Comment by rrosen** [comment]
I was seeing similar behavior.  In .eslintrc.js I had:
```
extends: ["eslint:recommended", "plugin:compat/recommended", "preact"],
```

I removed the preact entry and the checks to browsers not defined in the broweserslist config went away.

**Comment by amilajack** [comment]
@rrosen are you able to repro this in latest version?

**Comment by rrosen** [comment]
@amilajack I upgraded to `eslint-plugin-compat@3.7.0` and it's still happening

**Comment by amilajack** [comment]
@rrosen How are you defining your target browers?

**Comment by rrosen** [comment]
```
"browserslist": [
    "last 2 Chrome versions",
    "last 2 Firefox versions",
    "last 2 Edge versions",
    "last 3 Safari versions",
    "IE >= 11",
    "last 3 iOS versions",
    "last 3 ChromeAndroid versions"
  ]
```

**Comment by giovannetti-eric** [comment]
Hi,

I have a similar problem here but only in my editor:

```js
// .eslintrc.js
module.exports = {
  root: true,
  env: {
    browser: true,
    es2020: true,
    jest: true,
    node: true,
  },
  extends: [
    "eslint:recommended",
    "plugin:prettier/recommended",
    "plugin:compat/recommended",
  ],
  parserOptions: {
    sourceType: "module",
  },
};
```

```yaml
// .browserlistrc
defaults
not ie 11
```

```bash
$ npx browserslist
and_chr 81
and_ff 68
and_qq 10.4
and_uc 12.12
android 81
baidu 7.12
chrome 81
chrome 80
edge 81
edge 80
edge 18
firefox 76
firefox 75
firefox 74
firefox 68
ios_saf 13.4
ios_saf 13.3
ios_saf 12.2-12.4
kaios 2.5
op_mini all
op_mob 46
opera 68
opera 67
safari 13.1
safari 13
samsung 11.1
samsung 10.1
```

And I get errors in Visual Studio Code "Problems" tab:
"Object.assign() is not supported in IE 11 eslint(compat/compat)"

But everything ok if I run `npx eslint`.

I'm not sure if it's directly related to the current issue or not.

**Comment by Oblongmana** [comment]
+1, getting warnings for URLSearchParams not being supported in IE 11

Defining browserslist using package.json
```
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not ie > 0",
      "not opera > 0",
      "not ios_saf > 0",
      "not and_ff > 0",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
```

```
$ npx browserslist
and_chr 80
and_qq 1.2
and_uc 12.12
chrome 80
chrome 79
chrome 78
chrome 77
chrome 75
chrome 63
chrome 49
edge 18
firefox 73
firefox 72
safari 13
safari 12.1
samsung 11.1
samsung 10.1
samsung 9.2
```

**Comment by jrozbicki** [comment]
Can confirm, getting warrnings for:
```Promise is not supported in op_mini all, IE 11  compat/compat```

While `package.json`:
```
"browserslist": [
    "last 2 version",
    "not IE 11",
    "not op_mini all",
    "not dead"
  ],
```

and `npx browserslist` returns
```and_chr 84
and_ff 68
and_qq 10.4
and_uc 12.12
android 81
baidu 7.12
chrome 84
chrome 83
edge 84
edge 83
firefox 78
firefox 77
ios_saf 13.4-13.5
ios_saf 13.3
kaios 2.5
op_mob 46
opera 69
opera 68
safari 13.1
safari 13
samsung 12.0
samsung 11.1-11.2
```

I'm using:
`"eslint": "^6.8.0",`
`"eslint-plugin-compat": "^3.5.1",`

**Comment by LambdaZed** [comment]
Same problem.
`fetch is not supported in op_mini all, IE 11`
`Promise.all() is not supported in op_mini all, IE 11`

.browserslistrc
```
last 2 version
not IE 11
not op_mini all
not dead
```

npx browserslist
```
and_chr 87
and_ff 83
and_qq 10.4
and_uc 12.12
android 81
baidu 7.12
chrome 87
chrome 86
edge 87
edge 86
firefox 84
firefox 83
ios_saf 14.0-14.3
ios_saf 13.4-13.7
kaios 2.5
op_mob 59
opera 72
opera 71
safari 14
safari 13.1
samsung 13.0
samsung 12.0
```

.eslintrc.json
```
...
"extends": ["eslint:recommended", "airbnb-base", "plugin:compat/recommended"],
...
```

```
"eslint": "^7.18.0",
"eslint-plugin-compat": "^3.9.0",
```

**Comment by Gerben-T** [comment]
`v3.9.0` still has this issue

package.json
```
...
"devDependencies": {
	"eslint-plugin-compat": "^3.9.0",
}
"browserslist": [
	"defaults",
	"not ie <= 11",
	"not op_mini all"
]
```

`npx browserslist` output
```
and_chr 90
and_ff 87
and_qq 10.4
and_uc 12.12
android 90
baidu 7.12
chrome 90
chrome 89
chrome 88
chrome 87
edge 90
edge 89
edge 88
firefox 87
firefox 86
firefox 78
ios_saf 14.0-14.5
ios_saf 13.4-13.7
kaios 2.5
op_mob 62
opera 75
opera 74
safari 14
safari 13.1
samsung 13.0
samsung 12.0
```

![image](https://user-images.githubusercontent.com/26112463/116432160-b43de900-a848-11eb-95f1-edc60b61a0f9.png)

**Comment by neb-b** [comment]
This was fixed for me when I updated to 3.13.0 (not sure if versions before that were fixed, I was on 3.5.1)

**Comment by piotr-cz** [comment]
If you are extending [preact](https://github.com/preactjs/eslint-config-preact) config, the issue is caused from it setting own queries via `targets` property: https://github.com/preactjs/eslint-config-preact/blob/b38e3f37fc6b3f2ae45a64995bcff0a3ba3a3869/index.js#L53-L54

Workaround is to disable them in your eslint config:
```json
  "settings": {
    "targets": null
  }
```

**Comment by neongreen** [comment]
NB: I had the same issue with eslint-plugin-compat. I removed all plugins from the `extends` section, then added them back one by one, and.. it was gone. Not sure what happened there.

**Comment by HolgerJeromin** [comment]
I had the same issue. 
Fixed by deleting `.eslintcache` file. This is rebuild when the `.eslintrc.js` has changed, but my `browserslist` experiments were in `packages.json` only!

**Comment by JMaunel** [comment]
this seems to be a reoccurring issue. just encountered in the most recent version at time of this post.

`    "browserslist": [
        "last 2 iOS versions",
        "last 2 Chrome versions"
    ],`

`
 "eslint": "^8.54.0",
    "eslint-plugin-compat": "^4.2.0",
`
